sway/criteria.c

```diff
@@ -361,8 +361,17 @@ static char *get_focused_prop(enum criteria_token token) {
 			}
 		}
 		break;
-	case T_CON_ID: // These do not support __focused__
-	case T_CON_MARK:
+	case T_CON_ID:
+		if (view->swayc == NULL) {
+			return NULL;
+		}
+		size_t id = view->swayc->id;
+		int len = snprintf(NULL, 0, "%zu", id) + 1;
```

Comment:
Is this the standard way to compute the string's length?

Reply:
Yes. It's not a length (as in number of chars), it's a size (as in number of bytes that will get malloc'ed) so it probably makes more sense to name it `id_size`. Also it should be a `size_t`, not an `int`.

Reply:
Noted, thanks.

Reply:
>Also it should be a size_t, not an int.

*printf returns an int and so it makes sense to leave it as it is. It really is splitting hairs, though.

Reply:
fwiw I found that function after using snprintf like that myself, but for integer string length conversion we have this in `common/util.c` (declared in `include/util.h`):

```
int numlen(int n) {
        if (n == 0) {
                return 1;
        }
        return log10(n) + 1;
}
```

(It doesn't handle negative integers nor the full range of size_t, but both are fixable by just changing the prototype of the function. We only have one user though, it's a shame but it might be just as well to remove that function...)